### PR TITLE
K8SPS-258: check if backup is running

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPS-258](https://jira.percona.com/browse/K8SPS-258) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://perconadev.atlassian.net/browse/K8SPS-258

**DESCRIPTION**
---

**Problem:**
*If two `PerconaServerMySQLBackup` are created at the same time, both of them are set to the `Running` state, even when one of them is running the backup and the other is waiting.*

**Cause:**
After the `PerconaServerMySQLBackup` is created operator creates a backup job that checks on the source node, if the other backup is already running and waits. On the operator side, we just check if the backup job is running.

**Solution:**
*The operator should get information on the currently running backup from the source node of the backup. If this information doesn't correspond to the backup that the job is running, the `PerconaServerMySQLBackup` state should not be moved to `Running`.*

*It is done by adding a new handler for `GET` requests to the sidecar on the source node. The response of the handler will contain information about the currently running backup. The operator will send the request to the sidecar and check if the destination of the currently running backup is similar to the one that is currently being reconciled.*

*Additionally, this PR fixes an issue with the `delete-backup` finalizer. Previously, it didn't work when the cluster was running. The problem has been resolved by changing the sidecar port to match the default port used by the operator.*

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [x] Is an E2E test/test case added for the new feature/change?
- [x] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [x] Are all needed new/changed options added to default YAML files?
- [x] Did we add proper logging messages for operator actions?
- [x] Did we ensure compatibility with the previous version or cluster upgrade process?
- [x] Does the change support oldest and newest supported PS version?
- [x] Does the change support oldest and newest supported Kubernetes version?

[K8SPS-258]: https://perconadev.atlassian.net/browse/K8SPS-258?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| async-ignore-annotations | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/async-ignore-annotations.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/config.log) |
| config-router | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/config-router.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/demand-backup.log) |
| gr-demand-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/gr-demand-backup.log) |
| gr-demand-backup-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/gr-demand-backup-haproxy.log) |
| gr-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/gr-haproxy.log) |
| gr-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/gr-ignore-annotations.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/gr-init-deploy.log) |
| gr-one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/gr-one-pod.log) |
| gr-recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/gr-recreate.log) |
| gr-scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/gr-scaling.log) |
| gr-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/gr-self-healing.log) |
| gr-tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/gr-tls-cert-manager.log) |
| gr-users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/gr-users.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/haproxy.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/init-deploy.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/limits.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/monitoring.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/one-pod.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/operator-self-healing.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/recreate.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/scaling.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/sidecars.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/smart-update.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/tls-cert-manager.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/users.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/93c1fbc/version-service.log) |
| We run 30 out of 30|

commit: https://github.com/percona/percona-server-mysql-operator/pull/552/commits/93c1fbc40749f314cfca1417dd9ab8061a0e537d
image: `perconalab/percona-server-mysql-operator:PR-552-93c1fbc`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| async-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/async-ignore-annotations.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/config.log) |
| config-router | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/config-router.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/demand-backup.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/gr-demand-backup.log) |
| gr-demand-backup-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/gr-demand-backup-haproxy.log) |
| gr-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/gr-haproxy.log) |
| gr-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/gr-ignore-annotations.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/gr-init-deploy.log) |
| gr-one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/gr-one-pod.log) |
| gr-recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/gr-recreate.log) |
| gr-scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/gr-scaling.log) |
| gr-self-healing | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/gr-self-healing.log) |
| gr-tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/gr-tls-cert-manager.log) |
| gr-users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/gr-users.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/haproxy.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/init-deploy.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/limits.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/monitoring.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/one-pod.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/operator-self-healing.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/recreate.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/scaling.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/sidecars.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/smart-update.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/tls-cert-manager.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/users.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/21c6bee/version-service.log) |
| We run 30 out of 30|

commit: https://github.com/percona/percona-server-mysql-operator/pull/552/commits/21c6bee2164a557d038d8601b1b9a2a40ac5ab7d
image: `perconalab/percona-server-mysql-operator:PR-552-21c6bee`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| async-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/async-ignore-annotations.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/auto-config.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/config.log) |
| config-router | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/config-router.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/demand-backup.log) |
| gr-demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/gr-demand-backup.log) |
| gr-demand-backup-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/gr-demand-backup-haproxy.log) |
| gr-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/gr-haproxy.log) |
| gr-ignore-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/gr-ignore-annotations.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/gr-init-deploy.log) |
| gr-one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/gr-one-pod.log) |
| gr-recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/gr-recreate.log) |
| gr-scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/gr-scaling.log) |
| gr-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/gr-self-healing.log) |
| gr-tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/gr-tls-cert-manager.log) |
| gr-users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/gr-users.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/haproxy.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/init-deploy.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/limits.log) |
| monitoring | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/monitoring.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/one-pod.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/operator-self-healing.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/recreate.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/scaling.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/sidecars.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/smart-update.log) |
| tls-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/tls-cert-manager.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/users.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-552/7234d6f/version-service.log) |
| We run 30 out of 30|

commit: https://github.com/percona/percona-server-mysql-operator/pull/552/commits/7234d6fd405baebcd6524f6d19fbb6dfee631f8f
image: `perconalab/percona-server-mysql-operator:PR-552-7234d6f`

**Review by inelpandzic — APPROVED** [review]

File: pkg/xtrabackup/sidecarclient.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	"net/url"
+	"strconv"
+
+	"github.com/percona/percona-server-mysql-operator/pkg/mysql"
+	"github.com/pkg/errors"
+)
+
+type SidecarClient interface {
+	GetRunningBackupConfig(ctx context.Context) (*BackupConfig, error)
+	DeleteBackup(ctx context.Context, name string, cfg BackupConfig) error
```
Comment (egegunes) [inline pkg/xtrabackup/sidecarclient.go:18]: let's rename this to `DeleteBackupConfig`. its name is misleading now, it makes you think it'll delete the backup files

**Review by hors — APPROVED** [review]

**Pull request merged**